TypeCheck checking node_modules when running nuxi generate

### Environment

- Operating System: Linux
- Node Version:     v16.20.0
- Nuxt Version:     3.6.5
- Nitro Version:    2.5.2
- Package Manager:  npm@9.4.2
- Builder:          vite
- User Config:      typescript, runtimeConfig, modules
- Runtime Modules:  nuxt-icon@0.4.2
- Build Modules:    -

### Reproduction

Open up the Stackblitz. "ctrl + c" then run `npx nuxi generate`:
https://stackblitz.com/edit/github-stl7dc-qrtzu2?file=nuxt.config.ts

### Describe the bug

I installed nuxt-icon and enabled type checking.   Now running `npx nuxi generate` results in this error:

```

node_modules/nuxt-icon/dist/runtime/Icon.vue:29:54 - error TS2339: Property 'aliases' does not exist on type '{}'.

29 const iconName = computed(() => (appConfig.nuxtIcon?.aliases || {})[props.name] || props.name)
```

### Additional context

I tried adding `tsConfig` and excluding `node_modules` but this did not help.  

### Logs

```shell-script
❯ npx nuxi generate
Nuxi 3.6.5                                                                                                  5:07:22 PM
Nuxt 3.6.5 with Nitro 2.5.2                                                                                 5:07:22 PM

[5:07:23 PM]  WARN  Using experimental payload extraction for full-static output. You can opt-out by setting experimental.payloadExtraction to false.

ℹ Building client...                                                                                       5:07:24 PM
ℹ vite v4.3.9 building for production...                                                                   5:07:24 PM
ℹ ✓ 112 modules transformed.                                                                               5:07:27 PM
ℹ .nuxt/dist/client/manifest.json                   2.12 kB │ gzip:  0.38 kB                               5:07:28 PM
ℹ .nuxt/dist/client/_nuxt/entry.31621e1e.css        0.07 kB │ gzip:  0.09 kB                               5:07:28 PM
ℹ .nuxt/dist/client/_nuxt/IconCSS.6edc7bff.css      0.26 kB │ gzip:  0.17 kB                               5:07:28 PM
ℹ .nuxt/dist/client/_nuxt/error-500.c5df6088.css    1.95 kB │ gzip:  0.74 kB                               5:07:28 PM
ℹ .nuxt/dist/client/_nuxt/error-404.7fc72018.css    3.63 kB │ gzip:  1.12 kB                               5:07:28 PM
ℹ .nuxt/dist/client/_nuxt/IconCSS.cdf3b95e.js       0.86 kB │ gzip:  0.52 kB                               5:07:28 PM
ℹ .nuxt/dist/client/_nuxt/error-500.129a8e95.js     1.88 kB │ gzip:  0.99 kB                               5:07:28 PM
ℹ .nuxt/dist/client/_nuxt/error-404.f62b18ac.js     6.53 kB │ gzip:  2.91 kB                               5:07:28 PM
ℹ .nuxt/dist/client/_nuxt/entry.5eb7b0b7.js       127.30 kB │ gzip: 47.72 kB                               5:07:28 PM
ℹ ✓ built in 3.76s                                                                                         5:07:28 PM
✔ Client built in 3789ms                                                                                   5:07:28 PM
ℹ Building server...                                                                                       5:07:28 PM
ℹ vite v4.3.9 building SSR bundle for production...                                                        5:07:28 PM
transforming (50) node_modules/nuxt/dist/app/components/nuxt-link.jsnode_modules/nuxt-icon/dist/runtime/Icon.vue:12:38 - error TS2339: Property 'aliases' does not exist on type '{}'.

12 const aliases = appConfig?.nuxtIcon?.aliases || {}
                                        ~~~~~~~

node_modules/nuxt-icon/dist/runtime/Icon.vue:29:54 - error TS2339: Property 'aliases' does not exist on type '{}'.

29 const iconName = computed(() => (appConfig.nuxtIcon?.aliases || {})[props.name] || props.name)
                                                        ~~~~~~~

node_modules/nuxt-icon/dist/runtime/IconCSS.vue:7:38 - error TS2339: Property 'aliases' does not exist on type '{}'.

7 const aliases = appConfig?.nuxtIcon?.aliases || {}
                                       ~~~~~~~

node_modules/nuxt-icon/dist/runtime/IconCSS.vue:26:49 - error TS2339: Property 'size' does not exist on type '{}'.

26   if (!props.size && typeof appConfig.nuxtIcon?.size === 'boolean' && !appConfig.nuxtIcon?.size) {
```

Has there been any updates or workarounds on this?
I just ran into, what I believe is, the same issue when using the newest version of nuxt/ionic. 
(Nuxt v.3.10.1, Nuxt/Ionic v.0.13.0, Vue v.3.4.15)
```js
node_modules/@nuxtjs/ionic/dist/runtime/components/IonAnimation.vue:5:3 - error TS1484: 'Animation' is a type and must be imported using a type-only import when 'verbatimModuleSyntax' is enabled.

5   Animation,
    ~~~~~~~~~

node_modules/@nuxtjs/ionic/dist/runtime/components/IonAnimation.vue:6:3 - error TS1484: 'AnimationDirection' is a type and must be imported using a type-only import when 'verbatimModuleSyntax' is enabled.

6   AnimationDirection,
    ~~~~~~~~~~~~~~~~~~

node_modules/@nuxtjs/ionic/dist/runtime/components/IonAnimation.vue:7:3 - error TS1484: 'AnimationFill' is a type and must be imported using a type-only import when 'verbatimModuleSyntax' is enabled.

7   AnimationFill,
    ~~~~~~~~~~~~~

node_modules/@nuxtjs/ionic/dist/runtime/components/IonAnimation.vue:8:3 - error TS1484: 'AnimationKeyFrames' is a type and must be imported using a type-only import when 'verbatimModuleSyntax' is enabled.

8   AnimationKeyFrames,
    ~~~~~~~~~~~~~~~~~~
```

--- a/packages/nuxt/src/core/nuxt.ts
+++ b/packages/nuxt/src/core/nuxt.ts
@@ -129,6 +129,8 @@ async function initNuxt (nuxt: Nuxt) {
     if (nuxt.options.typescript.shim) {
       opts.references.push({ path: resolve(nuxt.options.buildDir, 'types/vue-shim.d.ts') })
     }
+    // Add shims for `#build/*` imports that do not already have matching types
+    opts.references.push({ path: resolve(nuxt.options.buildDir, 'types/build.d.ts') })
     // Add module augmentations directly to NuxtConfig
     opts.references.push({ path: resolve(nuxt.options.buildDir, 'types/schema.d.ts') })
     opts.references.push({ path: resolve(nuxt.options.buildDir, 'types/app.config.d.ts') })

--- a/packages/nuxt/src/core/templates.ts
+++ b/packages/nuxt/src/core/templates.ts
@@ -414,6 +414,7 @@ export const nuxtConfigTemplate: NuxtTemplate = {
   },
 }
 
+const TYPE_FILENAME_RE = /\.([cm])?[jt]s$/
 export const buildTypeTemplate: NuxtTemplate = {
   filename: 'types/build.d.ts',
   getContents ({ app }) {
@@ -424,12 +425,11 @@ export const buildTypeTemplate: NuxtTemplate = {
         continue
       }
 
-      const typeFilenames = [
-        file.filename.replace(/\.([cm])?[jt]s$/, '.d.$1ts'),
-        file.filename.replace(/\.([cm])?[jt]s$/, '.d.ts'),
-      ]
-      if (app.templates.some(f => f.filename && typeFilenames.includes(f.filename))) {
-        continue
+      if (TYPE_FILENAME_RE.test(file.filename)) {
+        const typeFilenames = new Set([file.filename.replace(TYPE_FILENAME_RE, '.d.$1ts'), file.filename.replace(TYPE_FILENAME_RE, '.d.ts')])
+        if (app.templates.some(f => f.filename && typeFilenames.has(f.filename))) {
+          continue
+        }
       }
 
       if (!file.filename.endsWith('.d.ts')) {

--- a/packages/nuxt/src/core/templates.ts
+++ b/packages/nuxt/src/core/templates.ts
@@ -415,13 +415,14 @@ export const nuxtConfigTemplate: NuxtTemplate = {
 }
 
 const TYPE_FILENAME_RE = /\.([cm])?[jt]s$/
+const DECLARATION_RE = /\.d\.[cm]?ts$/
 export const buildTypeTemplate: NuxtTemplate = {
   filename: 'types/build.d.ts',
   getContents ({ app }) {
     let declarations = ''
 
     for (const file of app.templates) {
-      if (file.write || !file.filename || file.filename.endsWith('.d.ts')) {
+      if (file.write || !file.filename || DECLARATION_RE.test(file.filename)) {
         continue
       }
 

--- a/packages/nuxt/src/core/templates.ts
+++ b/packages/nuxt/src/core/templates.ts
@@ -413,3 +413,30 @@ export const nuxtConfigTemplate: NuxtTemplate = {
     ].join('\n\n')
   },
 }
+
+export const buildTypeTemplate: NuxtTemplate = {
+  filename: 'types/build.d.ts',
+  getContents({ app }) {
+    let declarations = ''
+
+    for (const file of app.templates) {
+      if (file.write || !file.filename || file.filename.endsWith('.d.ts')) {
+        continue
+      }
+
+      const typeFilenames = [
+        file.filename.replace(/\.([cm])?[jt]s$/, '.d.$1ts'),
+        file.filename.replace(/\.([cm])?[jt]s$/, '.d.ts')
+      ]
+      if (app.templates.some(f => f.filename && typeFilenames.includes(f.filename))) {
+        continue
+      }
+
+      if (!file.filename.endsWith('.d.ts')) {
+        declarations += 'declare module ' + JSON.stringify(join('#build', file.filename)) + ';\n'
+      }
+    }
+
+    return declarations
+  }
+}

--- a/packages/nuxt/src/core/templates.ts
+++ b/packages/nuxt/src/core/templates.ts
@@ -416,7 +416,7 @@ export const nuxtConfigTemplate: NuxtTemplate = {
 
 export const buildTypeTemplate: NuxtTemplate = {
   filename: 'types/build.d.ts',
-  getContents({ app }) {
+  getContents ({ app }) {
     let declarations = ''
 
     for (const file of app.templates) {
@@ -426,7 +426,7 @@ export const buildTypeTemplate: NuxtTemplate = {
 
       const typeFilenames = [
         file.filename.replace(/\.([cm])?[jt]s$/, '.d.$1ts'),
-        file.filename.replace(/\.([cm])?[jt]s$/, '.d.ts')
+        file.filename.replace(/\.([cm])?[jt]s$/, '.d.ts'),
       ]
       if (app.templates.some(f => f.filename && typeFilenames.includes(f.filename))) {
         continue
@@ -438,5 +438,5 @@ export const buildTypeTemplate: NuxtTemplate = {
     }
 
     return declarations
-  }
+  },
 }

--- a/packages/nuxt/src/core/templates.ts
+++ b/packages/nuxt/src/core/templates.ts
@@ -432,9 +432,7 @@ export const buildTypeTemplate: NuxtTemplate = {
         }
       }
 
-      if (!file.filename.endsWith('.d.ts')) {
-        declarations += 'declare module ' + JSON.stringify(join('#build', file.filename)) + ';\n'
-      }
+      declarations += 'declare module ' + JSON.stringify(join('#build', file.filename)) + ';\n'
     }
 
     return declarations

